crypto3/libs/multiprecision/include/nil/crypto3/multiprecision/big_uint_impl.hpp

```diff
@@ -0,0 +1,1674 @@
+#pragma once
+
+// IWYU pragma: private; include "nil/crypto3/multiprecision/big_uint.hpp"
+
+#include <algorithm>
+#include <array>
+#include <bit>
+#include <boost/functional/hash.hpp>
+#include <cctype>
+#include <charconv>
+#include <climits>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <system_error>
+#include <type_traits>
+
+#include "nil/crypto3/multiprecision/detail/assert.hpp"
+#include "nil/crypto3/multiprecision/detail/config.hpp"
+#include "nil/crypto3/multiprecision/detail/endian.hpp"
+#include "nil/crypto3/multiprecision/detail/intel_intrinsics.hpp"
+#include "nil/crypto3/multiprecision/storage.hpp"
+
+namespace nil::crypto3::multiprecision {
+    template<std::size_t Bits>
+    class big_uint;
+
+    namespace detail {
+        constexpr bool is_valid_hex_digit(char c) {
```

Comment:
Constexpr on functions implies inline (https://en.cppreference.com/w/cpp/language/constexpr). Unnamed namespace would be worse because the parent namespace would be polluted.